mlflow/pyspark/ml/__init__.py

```diff
@@ -753,16 +753,60 @@ def log_post_training_metric(self, run_id, key, value):
 
 
 def _get_columns_with_unsupported_data_type(df):
+    from pyspark.ml.linalg import VectorUDT
+
     from mlflow.types.schema import DataType
 
     supported_spark_types = DataType.get_spark_types()
     unsupported_columns = []
     for field in df.schema.fields:
-        if field.dataType not in supported_spark_types:
+        if (field.dataType not in supported_spark_types) and not isinstance(
+            field.dataType, VectorUDT
+        ):
             unsupported_columns.append(field)
     return unsupported_columns
 
 
+def _check_or_set_model_prediction_column(spark_model, input_spark_df):
+    from pyspark.ml import PipelineModel
+
+    prediction_column = "prediction"
+    if isinstance(spark_model, PipelineModel) and spark_model.stages[-1].hasParam("outputCol"):
```

Comment:
Seems spark_model is always a PipelineModel, could we raise exception if it's not here? It's a bit weird that if this condition is False we directly return `prediction_column`

Reply:
I think no. We can also log a non-pipeline model as.

Reply:
> It's a bit weird that if this condition is False we directly return prediction_column

This is a designed behavior, i.e., for model logged to Mlflow artifacts, user must ensure its output column to be "prediction" column and mlflow will directly use the column as prediction result. ( https://github.com/mlflow/mlflow/pull/12758#discussion_r1689030042 ) .